src/Sylius/Bundle/AdminBundle/Twig/ErrorTemplateFinder/ErrorTemplateFinder.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Twig\ErrorTemplateFinder;
+
+use Sylius\Bundle\AdminBundle\SectionResolver\AdminSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Bundle\UiBundle\Twig\ErrorTemplateFinder\ErrorTemplateFinderInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Twig\Environment;
+
+class ErrorTemplateFinder implements ErrorTemplateFinderInterface
+{
+    public function __construct(
+        private SectionProviderInterface $sectionProvider,
+        private RequestStack $requestStack,
+        private Environment $twig,
+    ) {
+    }
+
+    public function findTemplate(int $statusCode): ?string
+    {
+        $section = $this->sectionProvider->getSection();
+
+        if ($section instanceof AdminSection && $this->isLoggedInAdmin()) {
+            $template = sprintf('@Twig/Exception/Admin/error%s.html.twig', $statusCode);
```

Comment:
We should move these twigs directly to Admin/Shop, right we're unnecessarily doing a back-and-forth between Twig's templates and one of the sections. Furthermore, the twigs that were added in the `templates/` directory won't be there in the end app, that's just the configuration of our internal application. And to top it off, since it's a direct change of base symfony logic (and their docs) regarding creating/overwriting error templates, we should add a note in the upgrade file.